tests/e2e/Services/Functions/FunctionsConsoleClientTest.php

```diff
@@ -95,10 +90,184 @@ public function testGetCollectionUsage(array $data)
+            'search' => $variableId
+        ]);
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertCount(1, $response['body']['variables']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testListVariables
+     */
+    public function testGetVariable(array $data)
+    {
+        $response = $this->client->call(Client::METHOD_GET, '/functions/' . $data['functionId'] . '/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals("APP_TEST", $response['body']['key']);
+        $this->assertEquals("TESTINGVALUE", $response['body']['value']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testGetVariable
+     */
+    public function testUpdateVariable(array $data)
+    {
+        $response = $this->client->call(Client::METHOD_PUT, '/functions/' . $data['functionId'] . '/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'key' => 'APP_TEST_UPDATE',
+            'value' => 'TESTINGVALUEUPDATED'
+        ]);
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+
+        $variable = $this->client->call(Client::METHOD_GET, '/functions/' . $data['functionId'] . '/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $variable['headers']['status-code']);
+        $this->assertEquals("APP_TEST_UPDATE", $variable['body']['key']);
+        $this->assertEquals("TESTINGVALUEUPDATED", $variable['body']['value']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testUpdateVariable
+     */
+    public function testDeleteVariable(array $data)
+    {
+        $response = $this->client->call(Client::METHOD_DELETE, '/functions/' . $data['functionId'] . '/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(204, $response['headers']['status-code']);
+
+        $response = $this->client->call(Client::METHOD_GET, '/functions/' . $data['functionId'] . '/variables', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals(0, sizeof($response['body']['variables']));
+
+        return $data;
```

Comment:
Could we add some more tests for failure cases here? 

- Create variable with key/value too large
- Get variable that doesn't exist
- Update variable that doesn't exist
- Update variable with key/value too large
- Delete variable that doesn't exist

Reply:
Added failure tests.